torchvision/datasets/lfw.py

```diff
@@ -0,0 +1,218 @@
+import os
+from typing import Any, Callable, Optional, Tuple
+from PIL import Image
+from torchvision.datasets import VisionDataset
+from torchvision.datasets.utils import check_integrity, download_and_extract_archive, download_url, verify_str_arg
+
+
+class _LFW(VisionDataset):
+
+    base_folder = 'lfw-py'
+    download_url_prefix = "http://vis-www.cs.umass.edu/lfw/"
+
+    file_dict = {
+        'original': ("lfw", "lfw.tgz", "a17d05bd522c52d84eca14327a23d494"),
+        'funneled': ("lfw_funneled", "lfw-funneled.tgz", "1b42dfed7d15c9b2dd63d5e5840c86ad"),
+        'deepfunneled': ("lfw-deepfunneled", "lfw-deepfunneled.tgz", "68331da3eb755a505a502b5aacb3c201")
+    }
+
+    def __init__(
+        self,
+        root: str,
+        train: bool,
+        image_set: str,
+        view: str,
+        transform: Optional[Callable] = None,
+        target_transform: Optional[Callable] = None,
+        download: bool = False,
+    ):
+        super(_LFW, self).__init__(os.path.join(root, self.base_folder),
+                                   transform=transform, target_transform=target_transform)
+
+        image_set = verify_str_arg(image_set.lower(), 'image_set', self.file_dict.keys())
+        images_dir, self.filename, self.md5 = self.file_dict[image_set]
+
+        self.view = verify_str_arg(view.lower(), 'view', ['people', 'pairs'])
+        self.split = "Train" if train else "Test"
+
+        if download:
+            self.download()
+
+        if not self._check_integrity():
+            raise RuntimeError('Dataset not found or corrupted.' +
+                               ' You can use download=True to download it')
+
+        self.images_dir = os.path.join(self.root, images_dir)
+
+    def loader(self, path: str) -> Image.Image:
+        with open(path, 'rb') as f:
+            img = Image.open(f)
+            return img.convert('RGB')
+
+    def _check_integrity(self):
+        fpath = os.path.join(self.root, self.filename)
+        fname = os.path.join(self.root, f"{self.view}Dev{self.split}.txt")
```

Comment:
Can't we also have checksums for these files? Should be only four distinct files, right?

Reply:
Yes
1. peopleDevTrain.txt
2. peopleDevTest.txt
3. pairsDevTrain.txt
4. pairsDevTest.txt

I'll try to add md5 checksum for these files as well.

Reply:
The md5 for these files (`{'pairs'|'people'}{''|'DevTrain'|'DevTest'}.txt`) is not provided on their website, so should I use the values calculated using [`calculate_md5`](https://github.com/pytorch/vision/blob/main/torchvision/datasets/utils.py#L53) function?

Ig rest all is done.